src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,161 @@
+        prepareFile();
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try (Scanner myScannerObj = new Scanner(filePath)) {
+            int lineNumber = 1;
+
+            while (myScannerObj.hasNext()) {
+                String line = myScannerObj.nextLine();
+                try {
+                    Task t = this.load(line);
+                    tasks.add(t);
+                } catch (DukeException e) {
+                    printWarning();
+                    System.err.println(lineNumber + ": " + e.getMessage());
+                }
+                lineNumber++;
+            }
+
+        } catch (IOException e) {
+            throw new DukeException("Unable to load tasks from task file.");
+        }
+
+        return tasks;
+    }
+
+    /**
+     * Saves tasks to the task file.
+     *
+     * @param tasks The list of tasks to be saved.
+     * @throws DukeException If there is an issue saving tasks.
+     */
+    public void saveTasks(ArrayList<Task> tasks) throws DukeException {
+        prepareFile();
+        try {
+            ArrayList<String> lines = new ArrayList<>();
+            for (Task t : tasks) {
+                lines.add(t.toFileString());
+            }
+            Files.write(filePath, lines, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.WRITE);
+        } catch (IOException e) {
+            throw new DukeException("Unable to save tasks to task file.");
+        }
+    }
+
+    /**
+     * Loads a task from a string representation.
+     *
+     * @param line The string representation of the task.
+     * @return The Task object created from the string representation.
+     * @throws DukeException If there is an issue loading the task.
+     */
+    public Task load(String line) throws DukeException {
+        String[] parts = line.split(" \\| ");
+
+        if (parts.length < 3 || !parts[1].matches("[01]")) {
+            throw new DukeException("Invalid task format. Please check the task file.");
+        }
+
+        String taskType = parts[0];
+        boolean isDone = Integer.parseInt(parts[1]) == 1;
+        String description = parts[2];
+
+        switch (taskType) {
+            case "T":
+                Task tT = new Todo(description);
+                tT.setDone(isDone);
+                return tT;
+            case "D":
+                Task tD = new Deadline(description, parts[3]);
+                tD.setDone(isDone);
+                return tD;
+            case "E":
+                String[] time = parts[3].split(" - ");
+                if (!(time.length == 2)) {
+                    throw new DukeException("Invalid time format. " +
+                            "Please use <datetime> - <datetime>");
+                }
+
+                Task tE = new Event(description, time[0], time[1]);
+                tE.setDone(isDone);
+                return tE;
+            default:
+                throw new DukeException("Invalid task type " + taskType);
+        }
```

Comment:
Maybe you should try to follow the switch/case convention in the [layout guide](https://se-education.org/guides/conventions/java/intermediate.html#layout)?